### ISSUE TITLE
Player leaving game before game starts error

### ISSUE THREAD

**Issue by p-thak** [cs340tabyu/cs340Winter2017#883]
When players join a game, but one player leaves before the game is started, once the game starts,  the game freezes so no other player can play their turn. This was tested on 3+ players.

**Comment by ribsalad** [comment]
Makes sense. Did you leave with the leave game button or by closing the client?

TODO: Jacob : When the player leaves the game, make sure the server does a better job cleaning up there person and that the initial turn state is not waiting for their pieces to be returned. This should happen when the leave game command tells the server a player left. This command calls remove player on the game. Have this method also ensure the things related to the initial start are set so the state knows to move on once all other players returned their cards.

**Comment by ribsalad** [comment]
To add to this, if one player leaves no player can even return their destination cards. Basically the server locks up in a weird spot and freezes all the clients where they are at. My bet is the game on the server either changes states, or is having a null pointer exception because it is trying to check each players destination cards to see if they are returned to go to the next player, but because one player left, it looks up the name and gets a null pointer exception when it tries to check the temporary dest cards on that player(which comes back as an error on the server). Another possibility is the state is getting changed prematurley on the server when a player leaves. Etierh way the command is sent of form the client fine but because there is no valid response so the client state is never given the appropriate even type to trigger it to change states on the client.

Jacob: Debug this on the server, specifically break point on the return destination card command on the server and test it after one player leaves the game in the initial sequence.

**Comment by tjacobhi** [comment]
That makes sense
